src/NUnitTestAdapter/Internal/AsyncMethodHelper.cs

```diff
@@ -24,7 +24,7 @@ public string GetClassNameForAsyncMethod(string className, string methodName)
             var definingType = targetAssembly.GetType(className);
             if (definingType == null) return null;
 
-            var method = definingType.GetMethod(methodName);
+            var method = definingType.GetMethods().Where(o => o.Name == methodName).OrderBy(o => o.GetParameters().Length).FirstOrDefault();
```

Comment:
I don't think it should be an issue, as you say we should never get in here in 2.0 since it doestn have async, but agree, need to check out this.